### ISSUE TITLE
Wrong shape of numerical DM created from an MX symbolic structure

### ISSUE THREAD

**Issue by mkotlyar** [casadi/casadi#1738]
Example:

``` python
import casadi as cs, casadi.tools as ct

M = 1   # If M is changed to something > 1, the error disappears.
N = 10
s = ct.struct_symMX([ct.entry('u', shape = (M, N))])
val = s(0)

assert s['u'].shape == (M, N)
assert val['u'].shape == (M, N)
```

The `assert` in the last line fails. The shape is `(10, 1)` instead of `(1, 10)`.

**Comment by jgillis** [comment]
Thanks for reporting! I guess I'm so accustomed to using column vectors, that the error never got exposed.

**Comment by mkotlyar** [comment]
Guys, any progress on that?
I still need inserting `if`s here and there to work it around.

**Comment by jgillis** [comment]
Raising priority

**Comment by jgillis** [comment]
Caused by #2154

**Comment by mkotlyar** [comment]
Is this issue really fixed? I just pulled from the develop branch 0b445a3328530145864c9c5e4fc2ca7db54b5ef1 and I still see the same problem.
